edge-modules/api-proxy-module/Cargo.toml

```diff
@@ -23,7 +23,7 @@ url = "2.1"
 
 azure-iot-mqtt = { path = "./rust-sdk/azure-iot-mqtt" }
 edgelet-client = { path = "../../mqtt/edgelet-client" }
-mqtt3 = { path = "../../mqtt/mqtt3" }
+mqtt3 = { git = "https://github.com/Azure/iotedge", rev = "4cd508" }
```

Comment:
instead of pinning to a specific revision directly in the `Cargo.toml`, consider pinning to a branch instead (i.e: `branch = "master"`), and managing the specific commit via `Cargo.lock`. This is the approach we use when working with `iot-identity-service` code in the `iotedge` repo (e.g: see https://github.com/Azure/iotedge/blob/master/edgelet/aziot-edged/Cargo.toml#L29-L38 + https://github.com/Azure/iotedge/blob/master/edgelet/Cargo.lock#L108-L119)

Reply:
once we merge it to `master` this pinning will stop working, right?

Reply:
hmm, that's a good question...

on a related note, what's the reason you're switching from using a relative path?

Reply:
I want to make incremental upgrade: first edgehub, then everything else.

Reply:
Ahhh, I see. Well, if it's only a temporary stop-gap while doing the incremental upgrade, then it _should_ be fine?

Reply:
I'd be nice to move it to a private `crates.io` registry and pin versions there...